### PR TITLE
MD Standard deduction

### PR DESCRIPTION
**Pull request by fedderw**
- [x] `make format && make documentation` has been run.

# New variable

- [x] Label field added
- [x] Documentation field added
- [x] Unit field added
- [ ] Default value field added if relevant
- [x] Variable name follows conventions
- [x] Unit test(s) added
- [ ] Integration test(s) added if relevant
- [ ] Issues this PR fixes linked

## What's changed

Description of the changes here.

# Bug fix

- [ ] Regression test added
- [ ] Regression test passing

## What this fixes and how it's fixed

Description of how this fix works goes here. Link any issues this PR fixes.

### CODE REVIEW

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    standard_deduction,
+)
+
+
+class md_standard_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MD standard deduction"
+    unit = USD
+    definition_period = YEAR
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:12]:
```suggestion
    definition_period = YEAR
    reference = "https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69"
```

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        filing_statuses = filing_status.possible_values
+        p = parameters(period).gov.states.md.tax.income.standard_deduction
+        # Start with md_agi
+        md_agi = tax_unit("md_agi", period)
+
+        # Caculate for single and separate depending on AGI.
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:21]:
```suggestion
        # Calculate for single and separate depending on AGI.
```

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/rate.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+metadata:
+  description: MD Standard deduction rate calculation 
+  period: year
+  name: md_standard_deduction_rate
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/rate.yaml:4]:
```suggestion
  unit: /1
  name: md_standard_deduction_rate
```

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+metadata:
+  description: MD Standard deduction minimum for single, married filing separately, or dependent taxpayers
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml:2]: descriptions go outside the metadata, labels go inside

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/rate.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+metadata:
+  description: MD Standard deduction rate calculation
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/rate.yaml:2]:
```suggestion
  description: MD maximum standard deduction as a share of AGI
```

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml (hunk 4 above)
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml:2]:
```suggestion
  description: MD minimum standard deduction for joint, head of household, or widow filers
```

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  name: md_single_separate_standard_deduction_max
+  period: year
+  reference:
+    - title: "MD Code, Tax - General, § 10-217 (c)(1)"
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/min.yaml:6]:
```suggestion
    - title: "MD Code, Tax - General, § 10-217 (c)(1), (d)(1)-(d)(3)"
```

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        filing_statuses = filing_status.possible_values
+        p = parameters(period).gov.states.md.tax.income.standard_deduction
+        # Start with md_agi
+        md_agi = tax_unit("md_agi", period)
+
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:20]:
```suggestion
        # Standard deduction is a percentage of AGI, bounded by a min/max by filing status.
```

File: openfisca_us/variables/gov/states/md/tax/income/md_deductions.py [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
     entity = TaxUnit
     label = "MD deductions"
     unit = USD
     definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        # Check if the tax_unit itemized on their federal returns:
+        tax_unit_itemizes = tax_unit("tax_unit_itemizes", period)
+        standard_deduction = tax_unit("md_standard_deduction", period)
+        deductions_if_itemizing = tax_unit_itemizes * tax_unit("taxable_income_deductions_if_itemizing", period)
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_deductions.py:16]: and subtract salt
```suggestion
        deductions_if_itemizing = tax_unit("taxable_income_deductions_if_itemizing", period)
```

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
+from openfisca_us.model_api import *
+from openfisca_us.variables.gov.irs.income.taxable_income.deductions.standard_deduction.standard_deduction import (
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:2]: remove these

File: openfisca_us/variables/gov/states/md/tax/income/subtractions/md_child_dependent_care_deduction.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+from openfisca_us.model_api import *
+
+
+class md_child_dependent_care_deduction(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/subtractions/md_child_dependent_care_deduction.py:4]: delete, along with other variables not related to standard deduction

File: openfisca_us/parameters/gov/states/md/tax/income/additions/additions.yaml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+description: ""
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/additions/additions.yaml:1]: can we delete this file?

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/max.yaml [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  name: md_single_separate_standard_deduction_max
+  period: year
+  reference:
+    - title: "MD Code, Tax - General, § 10-217 (c)(1)"
+      href: https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69
+    - title: "MD Code, Tax - General, § 10-217 (d)(1)-(d)(3)"
+      href: https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69
+    - title: "MD Code, Tax - General, § 10-217 (d)(1)-(d)(3)"
+      href: https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69
+    # TODO: Add newreference
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/max.yaml:12]: add reference for 2021 value - also for other indexed parameters here

File: openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/max.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+values:
+  2018-01-01: 4_500
+  2021-01-01: 4_700
+
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/standard_deduction/joint_head_widow/max.yaml:16]: remove extra line

File: openfisca_us/parameters/gov/states/md/tax/income/subtractions/child_dependent_care_deduction/max.yaml [hunk 14]
```diff
@@ -0,0 +1,10 @@
+description: Maximum child and dependent care expenses per dependent.
+metadata:
+  unit: currency-USD
+  name: md_child_dependent_care_deduction_max_expense
+  label: MD Maximum care expenses per dependent for child dependent care deduction.
+  period: year
+  reference:
+    -
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/subtractions/child_dependent_care_deduction/max.yaml:8]: add reference, or remove (isn't this the CDCC being done in #1093?)

File: openfisca_us/parameters/gov/states/md/tax/income/subtractions/child_dependent_care_deduction/max_dependents.yaml [hunk 15]
```diff
@@ -0,0 +1,8 @@
+description: MD maximum number of dependents qualifiable for child dependent care deduction.
+values:
+  2021-01-01: 2
+metadata:
+  unit: person
+  name: md_cdc_max_dependents
+  label: md_cdc maximum dependents
+  reference:
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/md/tax/income/subtractions/child_dependent_care_deduction/max_dependents.yaml:8]: same as above

File: openfisca_us/variables/gov/states/md/tax/income/additions/md_lump_sum_retirement_distribution.py [hunk 16]
```diff
@@ -0,0 +1,18 @@
+from openfisca_us.model_api import *
+
+
+class md_lump_sum_retirement_distribution(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/additions/md_lump_sum_retirement_distribution.py:4]: remove these additions from this pr

File: openfisca_us/variables/gov/states/md/tax/income/md_deductions.py [hunk 17]
```diff
@@ -7,3 +7,18 @@ class md_deductions(Variable):
     label = "MD deductions"
     unit = USD
     definition_period = YEAR
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_deductions.py:9]: add reference - is this actually how salt is removed? we'll need to follow the law here

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+    label = "MD standard deduction"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69"
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        filing_statuses = filing_status.possible_values
+        p = parameters(period).gov.states.md.tax.income.standard_deduction
+        # TODO fed agi in meantime
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:16]: remove todo - can leave an issue instead

File: openfisca_us/variables/gov/states/md/tax/income/additions/md_lump_sum_retirement_distribution.py (hunk 16 above)
Comment (fedderw) [reply openfisca_us/variables/gov/states/md/tax/income/additions/md_lump_sum_retirement_distribution.py:4]: Yikes. I really did some damage with my merges

File: openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    reference = "https://govt.westlaw.com/mdc/Document/NC8EB19606F6911E8A99BCF2C90B83D38?viewType=FullText&originationContext=documenttoc&transitionType=CategoryPageItem&contextData=(sc.Default)#co_anchor_I552E3B107DF711ECA8F2FF3A9E62BB69"
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        filing_statuses = filing_status.possible_values
+        p = parameters(period).gov.states.md.tax.income.standard_deduction
+        md_agi = tax_unit("md_agi", period)
+        # Standard deduction is a percentage of AGI, bounded by a min/max by filing status.
+        # Calculate for single and separate depending on AGI.
+        single_separate = max_(
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/md/tax/income/md_standard_deduction.py:19]: This works for now though in the future for reference we could do
```python
import numpy as np

single_separate = np.clip(p.rate * md_agi, p.single_separate.min, p.single_separate.max)
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**